contrib/partitioner/partitioner.py

```diff
@@ -95,116 +101,148 @@ def main():
         print('Data mount exists')
 
 
-    if len(usb_devs()) == 1:
-        if len(usb_partitions()) < 1:
-            try:
-                print("Running parted...")
-                os.system('/sbin/parted -s /dev/sda mkpart p ext4 3 100%');
-                print('Making first partition');
-                os.system('/sbin/mkfs.ext4 -F /dev/sda1');
-            except:
-                print('Error running parted');
-                sys.exit(1);
-        else:
-            print("Already partitioned")
-
-
-
-        if len(usb_partitions()) == 1:
-            try:
-                os.system('/bin/mount /dev/' + usb_partitions()[0] + ' /mnt/data')
-                # if .rekt exists or bitcoin directory doesnt exist (because the drive is a factory default)
-                # then wipe the drive and format it with EXT4
-                if os.path.exists('/mnt/data/.rekt') or not os.path.exists('/mnt/data/bitcoin'):
-                    print('REKT file exists OR bitcoin folder not found... So lets format it')
-                    # unmount before format
-                    os.system('/bin/umount /mnt/data')
-                    print('Initializing filesystem')
-                    os.system('/sbin/mkfs.ext4 -F /dev/' + usb_partitions()[0])
-                    # remount
-                    os.system('/bin/mount -t ext4 /dev/' + usb_partitions()[0] + ' /mnt/data')
-                    '''
-                    Get Size of SDA and partition info
-                    '''
-                    first_part = dev_size('sda') / (1000*1000)
-                    prune_setting = int(first_part / 2)
-
-                    if first_part < 512000:
-                        print("Pruning the config, because drive is less than 512 GB")
-                        os.system('/bin/sed -i "s/prune=550/prune=' + str(prune_setting) + '/g;" ' + str(homedirpath) + '/bitcoin/bitcoin.conf')
-                    else:
-                        print("Switching off pruning, and turn on txindex")
-                        os.system('/bin/sed -i "s/prune=550/#prune=550/g;" ' + str(homedirpath) + '/bitcoin/bitcoin.conf')
-                        os.system('/bin/sed -i "s/#txindex=1/txindex=1/g;" ' + str(homedirpath) + '/bitcoin/bitcoin.conf')
-
-                    '''
-                    Setup secrets, db, bitcoin, nginx, and lnd directory.. as a new install
-                    '''
-                    print('Setup secrets, db, bitcoin, nginx, tor and lnd directory.. as a new install')
+    # STEP to check If  no partitions found
+    if len(usb_partitions()) < 1:
+        try:
+            device_name=str(usb_partitions()[0])[:-1]
```

Comment:
I don't think this can ever work.

If `len(usb_partitions()) < 1` is true, then `usb_partitions()` is an empty list so `device_name=str(usb_partitions()[0])[:-1]` will fail because `usb_partitions()[0]` won't exist.

Reply:
That line will get triggered if there is no partitions.. that was one of the issues reported on telegram

Reply:
I see your point about ```device_name=str(usb_partitions()[0])[:-1]```

Reply:
I'll stick with sda for now...  but eventually switch to something better

https://github.com/nolim1t/umbrel-compose/blob/patch-2/contrib/partitioner/partitioner.py